### PR TITLE
Remove detection of wine and some others

### PR DESCRIPTION
**Pull request by huoyaoyuan**
Rationale:

Wine:
Linux and Mac users are supposed to use native .NET Core. After .NET Framework target was removed, wine should be also removed.

IsUnix:
Missing Android. Actually the member isn't used anywhere.

Is32Bit/Is64Bit:
They are ineffective. Caching them makes JIT fails to treat them as constants.
They are not used anywhere.
Their meanings are not clear. Ambiguous on OS/process, and x64/arm64.

### CODE REVIEW

File: osu.Framework/RuntimeInfo.cs [hunk 1]
```diff
@@ -20,11 +18,18 @@ public static class RuntimeInfo
         public static string GetFrameworkAssemblyPath() =>
             System.Reflection.Assembly.GetAssembly(typeof(RuntimeInfo)).Location;
 
-        public static bool Is32Bit { get; }
-        public static bool Is64Bit { get; }
+        [Obsolete("Use Environment.Is64Bit*, IntPtr.Size, or RuntimeInformation.*Architecture instead.")] // can be removed 20200430
+        public static bool Is32Bit => IntPtr.Size == 4;
+
+        [Obsolete("Use Environment.Is64Bit*, IntPtr.Size, or RuntimeInformation.*Architecture instead.")] // can be removed 20200430
+        public static bool Is64Bit => IntPtr.Size == 8;
```
Comment (bdach) [inline osu.Framework/RuntimeInfo.cs]: If checking `IntPtr.Size` is one of the suggested proposals for replacing these, then why are these being obsoleted, if it's the same thing? Seems a little strange.

Comment (huoyaoyuan) [reply osu.Framework/RuntimeInfo.cs]: I just tried not to make breaking change, thus keeping them to match current behavior with better performance.

File: osu.Framework/RuntimeInfo.cs [hunk 2]
```diff
@@ -20,11 +18,18 @@ public static class RuntimeInfo
...
+
         public static Platform OS { get; }
-        public static bool IsUnix => OS == Platform.Linux || OS == Platform.MacOsx || OS == Platform.iOS;
-        public static bool IsWine { get; }
+        public static bool IsUnix => OS != Platform.Windows;
```
Comment (jai-x) [inline osu.Framework/RuntimeInfo.cs]: Does this change behaviour in the case of Android?

Comment (bdach) [reply osu.Framework/RuntimeInfo.cs]: See PR description, this doesn't actually seem to have any users reading it at all.

Comment (jai-x) [reply osu.Framework/RuntimeInfo.cs]: Oh, never mind then

File: osu.Framework/RuntimeInfo.cs [hunk 3]
```diff
@@ -8,9 +8,7 @@ namespace osu.Framework
 {
     public static class RuntimeInfo
     {
-        [DllImport(@"kernel32.dll", CharSet = CharSet.Ansi, ExactSpelling = true, SetLastError = true)]
-        internal static extern IntPtr GetProcAddress(IntPtr hModule, string procName);
-
+        [Obsolete("This P/Invoke was accidentially exposed.")] // can be removed 20200430
```
Comment (smoogipoo) [inline osu.Framework/RuntimeInfo.cs]: It's fine to remove this straight up, I don't see it as a "documented" API. Technically none of this is that, and it's really only intended for internal use in the first place.

**Pull request merged**